charts/redpanda/values.go

```diff
@@ -293,7 +295,25 @@ type Monitoring struct {
 	EnableHttp2    *bool                   `json:"enableHttp2"`
 }
 
+// RedpandaResources encapsulates the calculation of the redpanda container's
+// [corev1.ResourceRequirements] and parameters such as `--memory`,
+// `--reserve-memory`, and `--smp`.
+// This calculation occurs in two "modes"
+// * "Implied" - When Limits and Requests are set.
+// * "Historic" - When Limits and Requests are not set.
+//
+// Implied mode (recommended) will respect the Limits and Requests as given and
+// calculate redpanda's CLI flags based off what's provided. Should additional
+// tuning be required, flags may be set directly through their appropriate
+// values.yaml option.
+//
+// Historic mode is the default which introduced a bespoke modeling of both
+// redpanda's CLI flags and container resources. See it's substructs for
+// further explanation.
```

Comment:
```suggestion
// This calculation supports two modes:
// * Explicit mode (recommended):  Activated when `Limits` and `Requests` are set.
//   In this mode, the CLI flags are calculated directly based on the provided 
//   `Limits` and `Requests`. This mode ensures predictable resource allocation
//   and is recommended for production environments. If additional tuning is required, 
//   the CLI flags can be manually overridden using their respective values in the Helm chart.
//
// * Legacy mode (default): Used when `Limits` and `Requests` are not set.
//   In this mode, the container resources and CLI flags are calculated using
//   built-in default logic, where 80% of the container's memory is allocated to Redpanda 
//   and the rest is reserved for system overhead. Legacy mode is intended for backward 
//   compatibility and less controlled environments.
//
// Explicit mode offers better control and aligns with Kubernetes best practices.
// Legacy mode is a fallback for users who have not defined `Limits` and `Requests`.
```

Reply:
'Explicit mode' is more descriptive as it emphasizes that users explicitly set requests and limits for control.

Legacy mode implies that it's backward 
compatible while also signaling that users should switch to explicit mode.